src/platform/packages/shared/kbn-config-schema/src/types/object_type.ts

```diff
@@ -227,8 +227,6 @@ export class ObjectType<P extends Props = any> extends Type<ObjectResultType<P>>
 
   /**
    * Return the schema for this object's underlying properties
-   *
-   * @internal should only be used internal for type reflection
    */
   public getPropSchemas(): P {
```

Comment:
@elastic/kibana-core I'm curious if you would be against this change.

I don't see this being much different than `extends` in how it handles the `props`. Maybe there is a need to return `{ ...this.props }` instead.

I thought about adding a `pick` function but the `defaultValue` types are more than I want to bit off, and the `Reference` type is strange.

Open to discussion I would not be blocked without this but it would be very nice to have if there is not strong opinion.

Reply:
I think the original PR introduced this with the intention of solving an internal problem and so tried to avoid the extra work of expanding the public contract to do so

Given that this is transparently returning the `props` you passed in I think this makes sense. Would you mind adding a test case(s) to `src/platform/packages/shared/kbn-config-schema/src/types/object_type.test.ts`.

Reply:
Good point, added tests in 89e532a9804f10e72e6390962b0cf3d7a45fba13. Let me know if there's anything else you'd like me to add.